collabora_online.module

```diff
@@ -116,15 +118,23 @@ function collabora_online_entity_operation(EntityInterface $entity): array {
     ],
   ];
 
-  if (
-    CoolUtils::canEditMimeType($type) &&
-    $media->access('edit in collabora')
-  ) {
-    $entries['collabora_online_edit'] = [
-      'title' => t("Edit in Collabora Online"),
-      'weight' => 50,
-      'url' => CollaboraUrl::editMedia($media),
-    ];
+  if ($media->access('edit in collabora')) {
+    /** @var \Drupal\collabora_online\Discovery\DiscoveryFetcherInterface $discovery_fetcher */
+    $discovery_fetcher = \Drupal::service(DiscoveryFetcherInterface::class);
+    try {
+      $discovery = $discovery_fetcher->getDiscovery();
+      $wopi_client_edit_url = $discovery->getWopiClientURL('edit', $type);
+      if ($wopi_client_edit_url !== NULL) {
+        $entries['collabora_online_edit'] = [
+          'title' => t("Edit in Collabora Online"),
+          'weight' => 50,
+          'url' => CollaboraUrl::editMedia($media),
+        ];
+      }
+    }
+    catch (CollaboraNotAvailableException $e) {
+      Error::logException(\Drupal::logger('cool'), $e);
```

Comment:
Instead of logging we could raise a warning, which would show up in the browser in a local development instance, and go to the logs in production.

Either way there is a risk to spam the logs.

Reply:
Already commented on this.